File: homework_1/8.digits_of_num/digits_of_num.js
const prompt = require('prompt-sync')()

let n = +prompt()

let max = 1, n_copy = n;

while(n != 0){
    n = Math.floor(n / 10);
    max *= 10;
}

while(max != 1){
    max /= 10;
    let digit = Math.floor((n_copy % (max * 10)) / max);
    console.log(digit);
}

File: homework_1/8.digits_of_num/digits_of_num.cpp
#include <iostream>
using namespace std;

int main() {
    int n, n_copy, max = 1;

    cin >> n;

    n_copy = n;

    while(n != 0){
        n /= 10;
        max *= 10;
    }

    while(max != 1){
        max /= 10;
        int digit = (n_copy % (max * 10)) / max;
        cout << digit << ' ';
    }
    
    cout << endl;

    return 0;
}